### PR TITLE
refactor(lidar_centerpoint): add training docs

### PR DESCRIPTION
**Pull request by kaancolak**
## Description

Add training documentation about training lidar CenterPoint model.

### Blocks merging of:
- https://github.com/autowarefoundation/autoware-documentation/pull/471

### Blocked by:
- https://github.com/autowarefoundation/mmdetection3d/pull/1

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/lidar_centerpoint/README.md [hunk 1]
```diff
@@ -80,6 +80,175 @@ ros2 launch lidar_centerpoint single_inference_lidar_centerpoint.launch.xml pcd_
...
+
+#### Install mmdetection3d
+
+**Step 1.** Install MMEngine, MMCV, and MMDetection using MIM
+
+```bash
+pip install -U openmim
+mim install mmengine
+mim install 'mmcv>=2.0.0rc4'
+mim install 'mmdet>=3.0.0rc5, <3.3.0'
```
Comment (Shin-kyoto) [inline perception/lidar_centerpoint/README.md:119]: @kaancolak 

Can you add dockerfile to make this environment in https://github.com/autowarefoundation/mmdetection3d.git?
We can make installation step shorter.

File: perception/lidar_centerpoint/README.md [hunk 2]
```diff
@@ -80,6 +80,175 @@ ros2 launch lidar_centerpoint single_inference_lidar_centerpoint.launch.xml pcd_
...
+```
+
+**Step 2.** Install mmdetection3d forked repository
+
+Introduced several valuable enhancements in our fork of the mmdetection3d repository.
+Notably, we've made the PointPillar z voxel feature input optional to maintain compatibility with the original paper.
+In addition, we've integrated a PyTorch to ONNX converter and a Tier4 Dataset format reader for added functionality.
```
Comment (Shin-kyoto) [inline perception/lidar_centerpoint/README.md:126]:
```suggestion
In addition, we've integrated a PyTorch to ONNX converter and a T4 format reader for added functionality.
```

Please check https://github.com/tier4/tier4_perception_dataset/blob/main/docs/t4_format_3d_detailed.md

File: perception/lidar_centerpoint/README.md [hunk 3]
```diff
@@ -80,6 +80,175 @@ ros2 launch lidar_centerpoint single_inference_lidar_centerpoint.launch.xml pcd_
...
+ln -s /PATH/TO/DATASET/ /PATH/TO/mmdetection3d/data/tier4_dataset/
+```
+
+##### Prepare dataset and evaluate trained model
+
+Create .pkl files for training, evaluation, and testing.
+
+```bash
+
+python tools/create_data.py Tier4Dataset --root-path data/sample_dataset/ --out-dir data/sample_dataset/ --extra-tag Tier4Dataset --version sample_dataset --annotation-hz 2
```
Comment (Shin-kyoto) [inline perception/lidar_centerpoint/README.md:194]:
```suggestion
python tools/create_data.py T4Dataset --root-path data/sample_dataset/ --out-dir data/sample_dataset/ --extra-tag T4Dataset --version sample_dataset --annotation-hz 2
```

**Review by Shin-kyoto — CHANGES_REQUESTED** [review]

File: perception/lidar_centerpoint/README.md (hunk 1 above)
Comment (kaancolak) [reply perception/lidar_centerpoint/README.md:119]: @Shin-kyoto -san, thank you for your advice!

Updated the current docker file with these installation steps. 

https://github.com/autowarefoundation/mmdetection3d/pull/1/files#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddcL1

Comment (Shin-kyoto) [reply perception/lidar_centerpoint/README.md:119]: @kaancolak 
Thank you so much!! Can you update this document to align with the environment setup procedures using Docker?

Comment (kaancolak) [reply perception/lidar_centerpoint/README.md:119]: @Shin-kyoto sure, I added instructions to here:
https://github.com/autowarefoundation/autoware.universe/pull/5570/files#diff-ef509aa43435872925d0134c14c088da6f51549d59e84cc5bd3e74fd4fa333f9R134

**Comment by xmfcx** [comment]
@kaancolak could you rebase this to latest main?

File: perception/lidar_centerpoint/README.md [hunk 4]
```diff
@@ -68,6 +68,198 @@ You can download the onnx format of trained models by clicking on the links belo
...
+python tools/create_data.py nuscenes --root-path ./data/nuscenes --out-dir ./data/nuscenes --extra-tag nuscenes
+```
+
+### Training CenterPoint with NuScenes Dataset
+
+#### Prepare the config file
+
+The configuration file that illustrates how to train the CenterPoint model with the NuScenes dataset is
+located at mmdetection3d/projects/AutowareCenterPoint/configs. This configuration file is a derived version of the
+centerpoint_pillar02_second_secfpn_head-circlenms_8xb4-cyclic-20e_nus-3d.py configuration file from mmdetection3D.
```
Comment (xmfcx) [inline perception/lidar_centerpoint/README.md:168]:
```suggestion
`centerpoint_pillar02_second_secfpn_head-circlenms_8xb4-cyclic-20e_nus-3d.py` configuration file from mmdetection3D.
```

**Comment by Shin-kyoto** [comment]
@kaancolak 

It seems that there is an issue with spell check. Could you please resolve it?

**Pull request merged**